server/models/services/ModelService.js

```diff
@@ -1,6 +1,48 @@
 const { isFunction, isEqualWith, sortBy } = require('lodash');
 const { sequelize } = require('..');
 
+/**
+ * Utility function to retry database operations that might deadlock
+ * @param {Function} operation - The database operation to perform
+ * @param {Object} options - Options for retry behavior
+ * @param {number} options.maxRetries - Maximum number of retry attempts (default: 3)
+ * @param {number} options.initialDelay - Initial delay in ms before retrying (default: 100)
+ * @param {number} options.backoffFactor - Factor to increase delay by on each retry (default: 2)
+ * @returns {Promise<*>} - Result of the operation
+ */
+const withDeadlockRetry = async (
```

Comment:
I have been finding that we occasionally have deadlock issues. This appears to resolve them but is orthogonal to the PR. Open to removing.